cpp/python/src/future.cpp

```diff
@@ -167,6 +179,124 @@ PyObject* future_set_exception(PyObject* future, PyObject* exception, const char
   return result;
 }
 
+PyObject* create_python_future_with_event_loop(PyObject* event_loop)
+{
+  PyObject* result = NULL;
+
+  PyGILState_STATE state = PyGILState_Ensure();
+
+  if (init_ucxx_python() < 0) {
+    if (!PyErr_Occurred()) PyErr_SetString(PyExc_RuntimeError, "could not allocate internals.");
+    goto finish;
+  }
+
+  result = PyObject_CallMethodObjArgs(event_loop, create_future_str, NULL);
+  if (PyErr_Occurred()) {
+    ucxx_trace_req("Error calling event loop `create_future`.");
+    PyErr_Print();
+  }
+
+finish:
+  PyGILState_Release(state);
+  return result;
+}
+
+PyObject* future_set_result_with_event_loop(PyObject* event_loop, PyObject* future, PyObject* value)
+{
+  PyObject* result              = NULL;
+  PyObject* set_result_callable = NULL;
+
+  PyGILState_STATE state = PyGILState_Ensure();
+
+  if (init_ucxx_python() < 0) {
+    if (!PyErr_Occurred()) PyErr_SetString(PyExc_RuntimeError, "could not allocate internals.");
+    goto finish;
+  }
+
+  set_result_callable = PyObject_GetAttr(future, set_result_str);
+  if (PyErr_Occurred()) {
+    ucxx_trace_req("Error getting future `set_result` method.");
+    PyErr_Print();
```

Comment:
I think you missed a jump here?
```suggestion
    PyErr_Print();
    goto finish;
```